self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryInvoices.java

```diff
@@ -184,8 +185,13 @@ public void pay(final int id) {
 
         /**
          * Stored invoices for this contract.
+         * We use a stream because we don't want to load in
+         * memory all the contract invoices.
+         * <br>
+         * In order to "reuse" them, since streams are one time use only,
+         * we wrap the stream in a supplier, .
          */
-        private final List<Invoice> contractInvoices;
+        private final Supplier<Stream<Invoice>> contractInvoices;
```

Comment:
Reversed back to stream.
If a new **invoice** is created in storage, the list will not be updated, unless we refresh by doing the whole "collect" ceremony from constructor.

Reply:
@criske Ok, let's leave it like that. Maybe I didn't completely understand the whole picture :D